src/utils/helpers/utilityFunctions.ts

```diff
@@ -35,3 +35,16 @@ export const substituteValues = (
   // Match ${...} using regex and replace ... with property from object
   return parameterisedString.replace(/(\${)(.*?)(})/gm, getObjectProperty)
 }
+
+/*
+ We are using ',' as a delimiter for url queries with multiple values. This means that values with *actual* commas in them get messed up. These functions can be used to remove commas before updating url query, then re-inserting them after extraction from url query
+*/
+
+// Use anything obscure than doesn't have special meaning in urls or regex
+const MAGIC_STRING = '~~~'
+
+export const removeCommas = (value: string) => value.replace(/,/g, MAGIC_STRING)
+export const replaceCommas = (value: string) => value.replace(new RegExp(MAGIC_STRING, 'g'), ',')
+
+export const removeCommasArray = (values: string[]) => values.map((value) => removeCommas(value))
+export const replaceCommasArray = (values: string[]) => values.map((value) => replaceCommas(value))
```

Comment:
The whole "commas within filter option values" thing was a major pain in my ass:
![2022-09-30 00 53 51](https://user-images.githubusercontent.com/5456533/193025148-97daf2c5-d4b1-4aec-9de3-27769f68ee0c.jpg)

Because multiple values are separated in the url using a comma, the query parser has no way to tell which commas are separators and which are actual commas!

So I've made these little helpers, and we replace the commas from strings at the last possible moment (just before being inserted into the url (with `updateQuery`) and replace them again immediate after parsing the query.

Open to suggestions for the MAGIC_STRING value. Needs to be something not too ugly that draws attention to itself in the url, but obscure enough that it's very unlikely to be used in an actual string. Oh, and not use any characters that have semantic significance in urls (e.g. `&`, `?`) or Regex (e.g. `\`, `.`,`^` etc)

Reply:
I do like the pipe `|` for that. And I think that was a smart solution.

Reply:
Yeah, that's not bad, but it would need to be at least a double one `||` as it's not unreasonable to expect a pipe in an actual data value.

You think it would be worth just putting this in preferences? And then if a country does find themselves using this character, then they can change it?

(I really would like to change it so that `+` is our global url delimiter -- that's more conventional, but that's a significant re-factor)

Reply:
Yes, I think it would be nice having that as a preferences - but probably wouldn't be used, so not to spent too long only if you think it's a quicker change. I wouldn't go with the significant refactor option for now LOL... But ideally the day we do this refactor is to also have the `+` substitution on the preferences (worth to put the idea on a issue though)